Source/WebKit/UIProcess/WebProcessCache.cpp

```diff
@@ -59,8 +63,38 @@ WebProcessCache::WebProcessCache(WebProcessPool& processPool)
     platformInitialize();
 }
 
-bool WebProcessCache::canCacheProcess(WebProcessProxy& process) const
+bool WebProcessCache::canCacheProcess(WebProcessProxy& process, CheckSuspendedState checkSuspendedState) const
 {
+    WEBPROCESSCACHE_RELEASE_LOG("canCacheProcess: memory usage: %lu", process.processID(), process.memoryUsage());
+
+#if PLATFORM(IOS)
+    if (WTF::ramUnused() < 256 * MB) {
+        WEBPROCESSCACHE_RELEASE_LOG("canCacheProcess: Not caching process because there is not enough free memory, free RAM: %lu", process.processID(), WTF::ramUnused());
+        return false;
+    }
+
+    if (process.hasMediaActivity()) {
+        WEBPROCESSCACHE_RELEASE_LOG("canCacheProcess: Not caching process because there is media activity.", process.processID());
+        return false;
+    }
+
+    if (checkSuspendedState == CheckSuspendedState::Yes && process.throttler().currentState() != ProcessThrottleState::Suspended) {
```

Comment:
I don't understand this check. The process might be in the middle of preparing for suspension and we would discard it as a result. This seems unfortunate.